testdata/compiler-pragmas.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"unsafe"
+	"time"
+)
+
+//go:linkname getCoroutine runtime.getCoroutine
+func getCoroutine() unsafe.Pointer
```

Comment:
I would prefer if this test only tested for the feature in question (`//go:linkname` with slightly mismatched types), without also relying on internals of the runtime. You can test for just this feature like this, for example (untested):

```go
var global = 3

//go:linkname foo foo
func foo(ptr unsafe.Pointer) {
    if ptr == unsafe.Pointer(&global) {
        println("pointer matches")
    } else {
        println("pointer mismatch!")
    }
}

//go:linkname foobar foo
func foobar(ptr *int)

func main() {
    foobar(&global)
}
```